lib/react.js

```diff
@@ -230,6 +230,8 @@ declare module react {
   declare var Component: typeof React$Component;
   declare var PureComponent: typeof React$PureComponent;
   declare var Element: typeof React$Element;
+  
+  declare type Node = string | number | React$Element<any> | Array<Node>;
```

Comment:
Except Fragment should just be `Iterable<Node>`. Otherwise it makes `Node` allowed to be empty which I think is not usually what you want.

Although since these nodes can end up rendering to whatever, it's not super helpful for types anyway.